docs/reference/contributing/connectivity/LoRaPortingGuide.md

```diff
@@ -0,0 +1,71 @@
+
+#### Porting a 3rd party LoRaWANStack 
+
+The vision driving Arm Mbed-OS entails that we have one operating system for a myriad of IoT technologies encompassing a multitude of devices or platforms. However, it does not limit the user to design something specific or tailored to his/her needs. In the light of this vision, Arm Mbed LoRaWAN APIs are desgined in such a way that a developer can totally replace the native Mbed-OS LoRaWAN stack with one of his/her own. 
+
+In this sub-section, we will be discussing how a 3rd party LoRaWAN stack can seamlessly provide services to existing Mbed-OS LoRaWAN applications or could reuse applications with minimal effort in terms of code. 
+<span class="notes">**Note:** The way a 3rd party LoRaWAN stack harnesses the powers of Arm Mbed-OS, i.e., synchronization methods (if using RTOS), timers, HAL etc is beyond the scope of this documentation.</span>
+
+[LoRaWANBase](https://github.com/ARMmbed/mbed-os/blob/master/features/lorawan/LoRaWANBase.h) class is a pure virtual class providing user facing APIs for a LoRaWAN stack.
+The native Arm Mbed LoRaWAN Stack implements `LoRaWANBase` as [LoRaWANInterface](https://github.com/ARMmbed/mbed-os/blob/master/features/lorawan/LoRaWANInterface.h) which then serves as a network interface for the application. Potentially, any developer or vendor can provide an implementation of `LoRaWANBase` and that particular implementation would serve as a network interface for the application. 
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/lora_base.png)<span>Figure 2. Inheriting from LoRaWANBase to provide portable APIs.</span></span>
+
+There can be many different flavours when it comes to devices supporting LoRaWAN technology. 
+
+**Case 1: Design based on LoRa Chipset**
+
+This design pattern follows the generic architecture with LoRa radio part based on the LoRa transceiver and antenna plus associated RF circuitry from any vendor. Developer chooses only LoRa radio chipset, LoRaWAN stack and application runs on the MCU. In this case, you can use any Mbed-enabled LoRaWAN stack, i.e., the native Mbed-OS LoRaWAN stack or your ported LoRaWAN stack.    
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/lora_radio_chipset.png)<span>Figure 3. Design based on LoRa RF chipset.</span></span>
+
+**Case 2: Design based on LoRa Module**
+
+A LoRa module means that the LoRa transceiver and an MCU  are bound together (not in same silicon package). This can be considered as a ready product with all the connections between Radio and MCU, plus antenna and other RF circuitry dealt with already by the vendor for the developer. For a developer it means a board that contains a target MCU and RF transceiver integrated on the board (not on chip). All the wirings are done already. We provide LoRaWAN stack and application on the MCU.
```

Comment:
I don't think we will provide application to anyone. 
Could be modified to something like: 
Mbed OS provides LoRaWAN stack and application developer can modify the example application for his/her needs.